src/Suave.Reloaded/Reloaded.fs

```diff
@@ -0,0 +1,175 @@
+﻿module Suave.Reloaded
+
+open System
+open System.Diagnostics
+open System.IO
+open Suave
+open Suave.Web
+open Suave.Types
+open Microsoft.FSharp.Compiler.Interactive.Shell
+open Hopac
+
+let help = """
+You should use this module by loading it into e.g. a console app, or a fsx file.
+
+You should move your app to a module.
+
+The fsx file and the Program.fs files should both spawn the server using this
+module which in turn calls the app module.
+"""
+
+[<Serializable>]
+type ScriptConfig =
+  { fsxPath    : FileInfo
+    modulePath : string
+    expression : string
+    homeFolder : DirectoryInfo option }
+
+type OnRunning = ScriptConfig -> unit
+
+let launchBrowser _ : OnRunning = Process.Start("http://localhost:8083") |> ignore
+
+let defaultScriptConfig =
+  let startTheBrowser 
+  { fsxPath    = FileInfo "App.fsx"
+    modulePath = "App"
+    expression = "app"
+    homeFolder = None
+  }
+
+module internal OutsideAppDomain =
+  open System.Text
+  open System.Text.RegularExpressions
+  open System.Threading
+  open System
+  open Suave.Logging
+
+  let private count = ref 0
+
+  // thanks http://www.superstarcoders.com/blogs/posts/executing-code-in-a-separate-application-domain-using-c-sharp.aspx
+  type Isolated<'t when 't :> MarshalByRefObject> =
+    val serialNo : int
+    val domain   : AppDomain
+    val typ      : Type
+    val value    : 't
+
+    new () =
+      let serialNo = Interlocked.Increment count
+      let domain = AppDomain.CreateDomain("Isolated#" + string serialNo, null, AppDomain.CurrentDomain.SetupInformation)
+      let typ = typeof<'t>
+      { serialNo = Interlocked.Increment(count)
+        domain   = domain
+        typ      = typ
+        value    = domain.CreateInstanceAndUnwrap(typ.Assembly.FullName, typ.FullName) :?> 't }
+
+    member x.Value = x.value
+
+    interface IDisposable with
+      member x.Dispose () =
+        AppDomain.Unload x.domain
+
+  let watcher (globPattern : string) : Hopac.Ch<FileSystemEventArgs> =
```

Comment:
One typically does not expose channel. Rather, one typically exposes alts or jobs or functions that return alts or jobs.